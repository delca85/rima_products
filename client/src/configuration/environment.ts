export const isBeta = () => Boolean(process.env.REACT_APP_ENV === 'beta');
export const isProduction = () => Boolean(process.env.REACT_APP_ENV === 'production');
