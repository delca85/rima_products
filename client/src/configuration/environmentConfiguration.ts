import { isBeta, isProduction } from './environment';

const graphql = {
  beta: 'http://localhost:7000/graphql',
  production: '/.netlify/functions/graphql',
};

export const getEnvironmentConfiguration = () => {
  if (isBeta()) {
    return {
      graphql: graphql.beta,
    };
  }
  if (isProduction()) {
    return {
      graphql: graphql.production,
    };
  }
  return {
    graphql: graphql.beta,
  };
};
