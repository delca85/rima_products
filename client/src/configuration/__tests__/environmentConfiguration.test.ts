import { isBeta, isProduction } from '../environment';
import { getEnvironmentConfiguration } from '../environmentConfiguration';

jest.mock('../environment');

describe('configuration by environment', () => {
  it('should return beta config by default', () => {
    isBeta.mockReturnValueOnce(false);

    const received = getEnvironmentConfiguration();

    expect(received).toMatchSnapshot();
  });

  it('should return beta config', () => {
    isBeta.mockReturnValueOnce(true);

    const received = getEnvironmentConfiguration();

    expect(received).toMatchSnapshot();
  });

  it('should return production config', () => {
    isProduction.mockReturnValueOnce(true);

    const received = getEnvironmentConfiguration();

    expect(received).toMatchSnapshot();
  });
});
