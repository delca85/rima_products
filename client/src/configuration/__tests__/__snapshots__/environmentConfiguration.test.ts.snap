// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configuration by environment should return beta config 1`] = `
Object {
  "graphql": "http://localhost:7000/graphql",
}
`;

exports[`configuration by environment should return beta config by default 1`] = `
Object {
  "graphql": "http://localhost:7000/graphql",
}
`;

exports[`configuration by environment should return production config 1`] = `
Object {
  "graphql": "/.netlify/functions/graphql",
}
`;
